service/src/main/java/bio/terra/cbas/controllers/MethodsApiController.java

```diff
@@ -479,19 +537,37 @@ private static MethodVersionDetails methodVersionToMethodVersionDetails(
         .description(methodVersion.description())
         .created(DateUtils.convertToDate(methodVersion.created()))
         .lastRun(initializeLastRunDetails(methodVersion.lastRunSetId()))
-        .url(methodVersion.url());
+        .url(methodVersion.url())
+        .branchOrTagName(methodVersion.branchOrTagName());
   }
 
-  private MethodDetails methodVersionToMethodDetails(MethodVersion methodVersion) {
+  private MethodDetails methodVersionToMethodDetails(
+      MethodVersion methodVersion, Boolean includeSourceDetails) {
     Method method = methodVersion.method();
+    GithubMethodSourceDetailsResponse sourceDetails;
+
+    if (includeSourceDetails != null) {
+      GithubMethodSource githubMethodSource = methodDao.getMethodSourceDetails(method.methodId());
+      sourceDetails =
+          new GithubMethodSourceDetailsResponse()
+              .methodId(githubMethodSource.methodId())
+              .path(githubMethodSource.path())
+              ._private(githubMethodSource._private())
+              .organization(githubMethodSource.organization())
+              .repository(githubMethodSource.repository());
```

Comment:
Clarifying Java question - this doesn't throw exception when values for these are `null` in database (which it would be until we backfill these values)?

Reply:
Currently `organization` is the only nullable column in the database. We haven't put any new methods into the db, so it hasn't thrown an exception. It would however if we tried to backfill these values and didn't extract them from the url.